compiler/modules/front/expr.bal

```diff
@@ -1388,18 +1388,26 @@ function codeGenCheckingTerminator(bir:BasicBlock bb, s:CheckingKeyword checking
 
 function codeGenFunctionCallExpr(ExprContext cx, bir:BasicBlock bb, s:FunctionCallExpr expr) returns CodeGenError|ExprEffect {
     string? prefix = expr.prefix;
-    bir:FunctionRef func;
-    bir:Register? funcRegister = ();
+    [bir:FunctionRef, bir:Register?]|[(), bir:Register] func;
     if prefix == () {
-        [func, funcRegister] = check genLocalFunctionRef(cx, expr.funcName, expr.qNamePos);
+        func = check genLocalFunction(cx, expr.funcName, expr.qNamePos);
     }
     else {
-        func = check genImportedFunctionRef(cx, prefix, expr.funcName, expr.qNamePos);
+        func = check genImportedFunction(cx, prefix, expr.funcName, expr.qNamePos);
     }
+    if func is [bir:FunctionRef, bir:Register?] {
```

Comment:
Need to separate out having signature from having a function ref.

Reply:
Fixed